### DISCUSSION TITLE
Side-effects when giving `agent` config to `useHive()`?

### DISCUSSION THREAD

**Discussion by 4nduril-ms** [graphql-hive/console#6052] [Help] [answered]
Hey there,

I have a question regarding the possibility of calling `useHive` and therefore `createHive` with an `agent` property in the config object.

Our goal is to provide our own logger with a specific log-level (debug) so that we no longer see Hive's standard `console.info()` output on our server logs when on a INFO level.

For one, I think it would be awesome to have that documented – at least I didn't find info other than in the source.
And for my actual question: I saw [in the code](https://github.com/graphql-hive/console/blob/main/packages/libraries/core/src/client/usage.ts#L80) that if no `agent` config is provided it defaults at some point to `{ maxSize: 1500 }`. I found no other obvious defaults. So what I did was to provide an agent config similar to this:
```javascript
useHive({
  agent: {
    logger: {
      info: ourLogger.debug,
      error: ourLogger.error
    },
    maxSize: 1500
  }
})
```
This works as we expected and we can't see any other implications. But I wanted to get some confirmation that this really has no other side-effects. I would really appreciate any feedback. Especially if I missed something in docs or code.

**Comment by n1ru4l** [comment, marked as answer]
Hey @4nduril-ms, excuse the late reply! We originally introduced this so we can write better unit tests for verifying the functionality. We usually do not recommend people to tweak the `maxSize` these as the default configuration is what we figured out to be best suited batch size for our usage reporting endpoint to consume.

In terms of overriding the logger, you are free to do that! I created a separate issue for tracking documentation of these configuration options: https://github.com/graphql-hive/console/issues/7114
